src/client/datascience/notebook/intellisense/pythonKernelCompletionProvider.ts

```diff
@@ -308,10 +308,7 @@ export function filterCompletions(
         return r;
     });
 
-    // If not inside of a string, filter out file names (things with a '.' in them or end with '/')
-    if (!insideString) {
```

Comment:
Because otherwise you end up with all the file names in the intellisense. There was an issue specifically for this.

Reply:
> names in the intellisense. There was an issue specifically for this.

Isn't this code already flaky enough, feels like just pure guess work here. We don't know whether a string containing a `.` or `/` is a path or not. It could be something valid.

> I think there's a logic bug further above. It's supposed to remove the . that jupyter returns.

Not sure what you mean, but removing this code makes it work as expected.

Reply:
> Not sure what you mean, but removing this code makes it work as expected.

Yes and breaks the original fix. You'll now get paths in all sorts of things.

Reply:
> We don't know whether a string containing a . or / is a path or not. It could be something valid.

Yes hence the reason it doesn't do this filtering inside of a string. Only outside.

Reply:
Ok, how about we only filter out stuff that contains `/` as those could be treated (assumed to be) as paths.
That way the items that contains `.` will still exist such as `df.column`.